xml_converter/src/icon_gen.cpp

```diff
@@ -459,36 +460,44 @@ vector<string> Icon::as_xml() const {
     return xml_node_contents;
 }
 
-waypoint::Icon Icon::as_protobuf(waypoint::Icon proto_icon) const {
-    waypoint::Trigger* trigger = new waypoint::Trigger();
-    bool set_trigger = false;
+waypoint::Waypoint Icon::as_protobuf() const {
```

Comment:
This should probably return a `waypoint::Icon` see previous comment on similar changes.